File: Shuttle.Access/Permissions.cs
namespace Shuttle.Access
{
    public static class Permissions
    {
        public static class View
        {
            public const string Role = "access://role/view";
            public const string Identity = "access://identity/view";
            public const string Permission = "access://permissions/view";
            public const string Sessions = "access://sessions/view";
        }

        public static class Register
        {
            public const string Role = "access://roles/register";
            public const string Identity = "access://identity/register";
            public const string Permission = "access://permission/register";
        }

        public static class Remove
        {
            public const string Role = "access://role/remove";
            public const string Identity = "access://identity/remove";
        }

        public static class Status
        {
            public const string Permission = "access://permission/status";
        }

        public static class Activate
        {
            public const string Identity = "access://identity/activate";
        }
    }
}

File: Shuttle.Access.Vue/src/permissions.ts
const Permissions = {
  Identities: {
    Activate: "access://identities/activate",
    Manage: "access://identities/manage",
    Register: "access://identities/register",
    Remove: "access://identities/remove",
    View: "access://identities/view",
  },
  Permissions: {
    Manage: "access://permissions/manage",
    Register: "access://permissions/register",
    Remove: "access://permissions/remove",
    View: "access://permissions/view",
  },
  Roles: {
    Manage: "access://roles/manage",
    Register: "access://roles/register",
    Remove: "access://roles/remove",
    View: "access://roles/view",
  },
  Sessions: {
    Manage: "access://sessions/manage",
    Remove: "access://sessions/remove",
    View: "access://sessions/view",
  },
  Tenants: {
    Manage: "access://tenants/manage",
    Register: "access://tenants/register",
    View: "access://tenants/view",
  },
};

export default Permissions;
